### DISCUSSION TITLE
Upgrade Outstatic button from within the Dashboard

### DISCUSSION THREAD

**Discussion by avitorio** [avitorio/outstatic#41] [Ideas]
As mentioned by @Summpot [here](https://github.com/avitorio/outstatic/issues/39), we could have an upgrade button within the Dashboard.

This update button would: read the package.json file, parse it, and update the version for `outstatic`. Then a commit could be created to trigger the deploy with the new version.

**Comment by avitorio** [comment]
I really like this idea, but one of the motivations behind Outstatic is for it to be accessible for non-coders.

My concern is the upgrade button can result in deploy fail, or some other unwanted side effect happening.

I'd be curious to hear about what the community thoughts are and possible approaches for this feature.

**Comment by thecodedrift** [comment]
I think it's an awesome idea, but will have a few problems

* deployments that use lockfiles will probably fail if the lockfile can't satisfy the package.json
* we could patch the lockfile similar to dependabot/renovate/etc, but that requires outstatic to know about more than just a package.json file
* this could be solved by users running one of these tools such as dependabot/renovate/etc, which would give you other benefits like preview deploys from vercel/netlify

Not saying we can't do an upgrade button, just that editing package.json alone probably isn't enough.

**Reply by avitorio** [reply]
You're right. This would be a powerful feature, and we can eventually work on it, let's put a hold on it for now.
